bugsnag-plugin-android-ndk/src/test/cpp/test_bsg_event.c

```diff
@@ -297,6 +297,44 @@ TEST test_event_unhandled(void) {
     free(event);
     PASS();
 }
+
+TEST test_event_metadata(void) {
+    bugsnag_event *event = init_event();
+    bugsnag_event_add_metadata_string(event, "str", "foo", "hello");
+    bugsnag_event_add_metadata_string(event, "str", "foo2", "hello again");
+    bugsnag_event_add_metadata_double(event, "double", "foo", 5.9);
+    bugsnag_event_add_metadata_bool(event, "bool", "foo", true);
+
+    // check key exists
+    ASSERT_EQ(BSG_NONE_VALUE, bugsnag_event_has_metadata(event, "non-existent", "foo"));
+    ASSERT_EQ(BSG_CHAR_VALUE, bugsnag_event_has_metadata(event, "str", "foo"));
+    ASSERT_EQ(BSG_NUMBER_VALUE, bugsnag_event_has_metadata(event, "double", "foo"));
+    ASSERT_EQ(BSG_BOOL_VALUE, bugsnag_event_has_metadata(event, "bool", "foo"));
+
+    // test return value for missing key
+    char dst[64] = "";
+    bugsnag_event_get_metadata_string(event, "non-existent", "foo", dst);
+    ASSERT_STR_EQ("", dst);
+    ASSERT_EQ(false, bugsnag_event_get_metadata_bool(event, "non-existent", "foo"));
+    ASSERT_EQ(0.0, bugsnag_event_get_metadata_double(event, "non-existent", "foo"));
+
+    // test return value for present key
+    bugsnag_event_get_metadata_string(event, "str", "foo", dst);
+    ASSERT_STR_EQ("hello", dst);
+    ASSERT_EQ(5.9, bugsnag_event_get_metadata_double(event, "double", "foo"));
+    ASSERT_EQ(true, bugsnag_event_get_metadata_bool(event, "bool", "foo"));
+
+    // test clearing values
+    bugsnag_event_clear_metadata_section(event, "str");
+    ASSERT_EQ(BSG_NONE_VALUE, bugsnag_event_has_metadata(event, "str", "foo"));
```

Comment:
For completeness, maybe test clearing a value with `bugsnag_event_clear_metadata` and also assert that other keys still exist in the section?

Reply:
Added an extra assertion here.